insights/archive_functions.py

```diff
@@ -0,0 +1,26 @@
+import re
+import shutil
+import tarfile
+
+
+def get_system_id(self):
+    """ Checking system id from archive """
+    tar = tarfile.open(self.archive_location, 'r')
+    tar.extractall()
+    self.machine_id = tar.extractfile(self.machine_id_path)
```

Comment:
pass the machine_id_path parameter in the definition of the function and make a relevant call, same goes for all the other variables being used in the respective functions.